src/Board.java

```diff
@@ -244,7 +277,10 @@ public boolean isInCheck(Color color) {
                 current.row = i;
                 current.column = j;
                 if (this.getPiece(current) != null && this.getPiece(current).color != color) {
-                    return this.getPiece(current).isValidMove(current, dest, this);
+                    check = this.getPiece(current).isValidMove(current, dest, this);
+                    if (check) {
+                        return check;
+                    }
```

Comment:
feel like this is more obvious: if it's true, return true

```suggestion
                    if (check) {
                        return true;
                    }
```